extensions/src/test/java/org/pentaho/platform/web/servlet/UploadFileUtilsTest.java

```diff
@@ -58,4 +60,16 @@ public void testCheckExtensions() throws Exception {
     testUtils.setAllowsNoExtension( true );
     assertTrue( testUtils.checkExtension( "", false ) );
   }
+
+  @Test
+  public void testGetAllExtensions() throws Exception {
+    UploadFileUtils testUtils = new UploadFileUtils( null );
+    assertNull( testUtils.removeFileName( null ) );
+    assertEquals( "",  testUtils.removeFileName( "" ) );
+    assertEquals( "",  testUtils.removeFileName( "aaaa" ) );
+    assertEquals( "",  testUtils.removeFileName( " bb " ) );
+    assertEquals( "a",  testUtils.removeFileName( "b.a" ) );
+    assertEquals( "b.a",  testUtils.removeFileName( "c.b.a" ) );
+    assertEquals( "c.b.a",  testUtils.removeFileName( "d.c.b.a" ) );
```

Comment:
Shouldn't the extension be just `.a` instead?

Reply:
No, we need ALL extensions!